MonoGame.Framework/Random.cs

```diff
@@ -0,0 +1,254 @@
+
+            #region Int8 Next methods
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int8 (byte) that is less than the specified maximum.
+            /// </summary>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int8 (byte)</returns>
+            public byte Next(byte max)
+            {
+                return (byte)Remap(UInt64ToInt8(XORShift64Star()), 0, byte.MaxValue, 0, max);
+            }
+
+            /// <summary>
+            /// Returns a random integer of type Int8 (byte) that is within a specified range.
+            /// </summary>
+            /// <param name="min">Minimum possible value</param>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int8 (byte)</returns>
+            public byte Next(byte min, byte max)
+            {
+                return (byte)Remap(UInt64ToInt16(XORShift64Star()), 0, byte.MaxValue, min, max);
+            }
+
+            #endregion
+
+            #region Int16 Next methods
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int16 (short) that is less than the specified maximum.
+            /// </summary>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int16 (short)</returns>
+            public short Next(short max)
+            {
+                return (short)Remap(UInt64ToInt16(XORShift64Star()), short.MinValue, short.MaxValue, 0, max);
+            }
+
+            /// <summary>
+            /// Returns a random integer of type Int16 (short) that is within a specified range.
+            /// </summary>
+            /// <param name="min">Minimum possible value</param>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int16 (short)</returns>
+            public short Next(short min, short max)
+            {
+                return (short)Remap(UInt64ToInt16(XORShift64Star()), short.MinValue, short.MaxValue, min, max);
+            }
+
+            #endregion
+
+            #region Int32 Next methods
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int32 (int).
+            /// </summary>
+            /// <returns>Random integer of type Int32 (int)</returns>
+            public int Next()
+            {
+                return (int)Remap(UInt64ToInt32(XORShift64Star()), int.MinValue, int.MaxValue, 0, int.MaxValue);
+            }
+
+            /// <summary>
+            /// Returns a non-negative random integer of type Int32 (int) that is less than the specified maximum.
+            /// </summary>
+            /// <param name="max">Maximum possible value</param>
+            /// <returns>Random integer of type Int32 (int)</returns>
+            public int Next(int max)
```

Comment:
Maybe it is worth changing these to `NextInt`.  For example this code:
```
var v = myRandom.Next(100);
```
Is that `Next(byte)` or `Next(int)` or `Next(long)`?  Of course experienced devs know it will be `int`, but it does make it harder for developers to understand the code at a glance.